lib/new_relic/local_environment.rb

```diff
@@ -142,10 +142,9 @@ def check_for_puma
     end
 
     def check_for_falcon
-      return unless defined?(::Falcon::Server) &&
-        NewRelic::LanguageSupport.object_space_usable?
-
-      @discovered_dispatcher = :falcon if find_class_in_object_space(::Falcon::Server)
+      if defined?(::Falcon::Server) && File.basename($0) == 'falcon'
+        @discovered_dispatcher = :falcon
+      end
```

Comment:
I'd prefer `$PROGRAM_NAME` and a one-liner if we can fit into 120 cols:

```suggestion
      @discovered_dispatcher = :falcon if defined?(::Falcon::Server) && File.basename($PROGRAM_NAME) == 'falcon'
```